libs/lb/lb-rs/src/service/activity_service.rs

```diff
@@ -31,43 +33,61 @@ impl<Client: Requester, Docs: DocumentService> CoreState<Client, Docs> {
         Ok(())
     }
 
-    fn normalize(&mut self, docs: &mut [DocActivityMetrics]) {
+    pub(crate) fn normalize_activity_metrics(
```

Comment:
Not clear to me why you turned this into a `HashMap` the resulting code seems worse

Reply:
I used a `HashMap` since it allowed me to get ids in quickly in search service, instead of sequentially searching for them. But on second thought, I can just turn the original `Vec` into a `HashMap` within search service, and make it a part of `SearchableFileInfo`.